### ISSUE TITLE
When sending Test Invoice, Bitmarck replies with validation errors

### ISSUE THREAD

**Issue by sticksen** [alex-therapeutics/diga-api-client#31]
## Problem

My setup is the following (it´s Kotlin, but I guess it´s understandable):
```
    @Singleton
    fun digaInformation(): DigaInformation = DigaInformation.builder()
        .digaName("diga-common-name")
        .digaId("12345")
        .manufacturingCompanyName("Some GmbH")
        .manufacturingCompanyIk("123456789")
        .netPricePerPrescription(BigDecimal(10))
        .applicableVATpercent(BigDecimal(19))
        .manufacturingCompanyVATRegistration("DE 1234567889")
        .contactPersonForBilling(
            DigaInformation.ContactPersonForBilling.builder()
                .fullName("John Doe")
                .phoneNumber("+49 175 1232345")
                .emailAddress("doe@some.com")
                .build()
        )
        .companyTradeAddress(
            DigaInformation.CompanyTradeAddress.builder()
                .adressLine("Diga Street 1")
                .postalCode("34553")
                .city("Digatown")
                .countryCode("DE")
                .build()
        )
        .build()
```
The lib is called in this manner:
```
override fun redeemTest(code: String, insurance: String): DigaInvoiceResponse {
        val invoice = DigaInvoice.builder()
            .digavEid("12345000")
            .invoiceId("1")
            .validatedDigaCode(DigaApiTestCode.VALID.code)
            .build()

        return digaApiClient
            .sendTestInvoiceRequest(invoice, insurance)
    }
```
I am calling the above function with insurance `CH`.

The answer is: https://gist.github.com/sticksen/eafa8e9e18ba92f3e99dbd4a46dc8473

## Solution

There seems to be an issue with the calculation of the total sum and / or taxing. In `DigaXmlJaxbRequestWriter`, line 212 I can also see a `TODO` that is maybe connected to this issue?

**Comment by fongie** [comment]
Thanks for opening the issue! I will look into it when I can, but of course if you see where the bug is you are welcome to submit a PR for it =)

**Comment by fongie** [comment]
closed in #32
